cosmo_tester/framework/util.py

```diff
@@ -485,3 +487,32 @@ def is_community():
         )
 
     return image_type in community_image_types
+
+
+@contextmanager
+def set_client_tenant(manager, tenant):
+    if tenant:
+        original = manager.client._client.headers[CLOUDIFY_TENANT_HEADER]
+
+        manager.client._client.headers[CLOUDIFY_TENANT_HEADER] = tenant
+
+    try:
+        yield
+    except:
+        raise
+    finally:
+        if tenant:
+            manager.client._client.headers[CLOUDIFY_TENANT_HEADER] = original
+
+
+def get_test_tenant(test_param, manager, cfy):
+    if is_community():
+        tenant = 'default_tenant'
+        # It is expected that the plugin is already uploaded for the
+        # default tenant
+    else:
+        tenant = test_param
+        cfy.tenants.create(tenant)
+        manager.upload_plugin('openstack_centos_core',
```

Comment:
It's a little odd that an action like uploading a plugin to the manager would be performed in an innocuously sounding func like `get_test_tenant `

Reply:
Yeah, I was slightly irked by that as well, but it also doesn't feel right as 'prepare_test_tenant', because if you're running community then it just returns one... but yes, I'll change it, probably to prepare_and_get_test_tenant. It's a bit more verbose but it does at least strongly imply that it's doing more than just retrieving some information.